lib/fs.js

```diff
@@ -1469,16 +1481,122 @@ fs.unwatchFile = function(filename, listener) {
   }
 };
 
-// Regexp that finds the next partion of a (partial) path
-// result is [base_with_slash, base], e.g. ['somedir/', 'somedir']
-const nextPartRe = isWindows ?
-  /(.*?)(?:[\/\\]+|$)/g :
-  /(.*?)(?:[\/]+|$)/g;
+// Functions that find the next portion of a (partial) path up to and including
+// a possible path separator (e.g. 'foo' or 'foo/')
+function findNextPartWin32(path, start) {
+  var i = start;
+  for (; i < path.length; ++i) {
+    const code = path.charCodeAt(i);
+    if (code === 47/*/*/ || code === 92/*\*/)
```

Comment:
I was mainly keeping things consistent with the path perf PR. I don't really have a strong preference either way, but I do think that having the literals next to the character codes is easier to visually scan.